pkg/iter/sc_finder.go

```diff
@@ -216,3 +222,81 @@ func getLabelsFromObject(obj runtime.Object) (map[string]string, bool) {
 	}
 	return nil, false
 }
+
+// getSubclusterSandboxStatusMap returns a map that contains sandboxed
+// subclusters
+func (m *SubclusterFinder) getSubclusterSandboxStatusMap() map[string]string {
+	scMap := map[string]string{}
+	for i := range m.Vdb.Status.Sandboxes {
+		sb := &m.Vdb.Status.Sandboxes[i]
+		for _, sc := range sb.Subclusters {
+			scMap[sc] = sb.Name
+		}
+	}
+	return scMap
+}
+
+// getVdbSubclusters returns the subclusters that are in the vdb
+func (m *SubclusterFinder) getVdbSubclusters(sandbox string) []*vapi.Subcluster {
+	if sandbox != vapi.MainCluster {
+		return m.getSandboxedSubclusters(sandbox)
+	}
+	return m.getMainSubclusters()
+}
+
+// getMainSubclusters returns the subclusters that are not part
+// of any sandboxes
+func (m *SubclusterFinder) getMainSubclusters() []*vapi.Subcluster {
+	subclusters := []*vapi.Subcluster{}
+	scMap := m.getSubclusterSandboxStatusMap()
+	for i := range m.Vdb.Spec.Subclusters {
+		sc := &m.Vdb.Spec.Subclusters[i]
+		if _, ok := scMap[sc.Name]; !ok {
+			subclusters = append(subclusters, sc)
+		}
+	}
+	return subclusters
+}
+
+// getSandboxedSubclusters returns the subclusters that belong to the given
+// sandbox
+func (m *SubclusterFinder) getSandboxedSubclusters(sandbox string) []*vapi.Subcluster {
+	subclusters := []*vapi.Subcluster{}
+	for i := range m.Vdb.Status.Sandboxes {
+		sb := &m.Vdb.Status.Sandboxes[i]
+		if sb.Name != sandbox {
+			continue
+		}
+		for _, scName := range sb.Subclusters {
+			subclusters = append(subclusters, m.Subclusters[scName])
+		}
+	}
+	return subclusters
+}
+
+// filterPods returns a pod list without main cluster pods if a non-empty sandbox named is
+// passed in, or pod list with only main cluster pods if no sandbox name is passed
+func (m *SubclusterFinder) filterPods(oldPods *corev1.PodList, sandbox string) *corev1.PodList {
+	newPods := []corev1.Pod{}
+	scMap := m.getSubclusterSandboxStatusMap()
+	for i := range oldPods.Items {
+		pod := oldPods.Items[i]
+		isSandbox := false
+		sbName := vapi.MainCluster
+		sc, isSCNode := pod.Labels[vmeta.SubclusterNameLabel]
```

Comment:
Are there cases where the pod won't have the subcluster name label?

Reply:
The initial list contains the pods owned by the operator, I know that's a stretch but I thought a pod owned by the operator does not mean it belongs/has belonged to a sc. If you think it can confuse I can remove it.

Reply:
We should already omit pods that aren't part of a subcluster. In `buildObjList`, see the check for `hasSubclusterNameLabel`. We may be able to simplify this a bit if we assume all pods will have the label you are after.